webapp/src/components/Vault/VaultActionsForm/Migration/VaultV2MigrationForm.tsx

```diff
@@ -52,52 +60,238 @@ const PillButton = styled.div`
   border-radius: 100px;
 `;
 
+const MigrationFormExtraContainer = styled.div`
+  background: ${colors.primaryText}0a;
+  padding: 32px 24px 16px 24px;
+  margin-top: -16px;
+  border: ${theme.border.width} ${theme.border.style} ${colors.border};
+  border-radius: ${theme.border.radius};
+  text-align: center;
+  z-index: 0;
+`;
+
+const MigratinFormModeList = ["options", "migrate"] as const;
+type MigrationFormMode = typeof MigratinFormModeList[number];
+
 interface VaultV2MigrationFormProps {
   vaultOption: VaultOptions;
   vaultAccount: VaultAccount;
+  onFormSubmit: () => void;
 }
 
 const VaultV2MigrationForm: React.FC<VaultV2MigrationFormProps> = ({
   vaultOption,
   vaultAccount,
+  onFormSubmit,
 }) => {
-  const asset = getAssets(vaultOption);
+  /** V1 Vault data */
+  const { deposits, vaultMaxWithdrawAmount, asset, decimals } =
+    useVaultData(vaultOption);
   const color = getVaultColor(vaultOption);
+  const { handleActionTypeChange, handleMaxClick } =
+    useVaultActionForm(vaultOption);
+
+  const [mode, setMode] = useState<MigrationFormMode>(MigratinFormModeList[0]);
+
+  const [weeklyMigrationAmount, weeklyMigrationAmountLimit] = useMemo((): [
+    BigNumber,
+    BigNumber
+  ] => {
+    const vaultCollaterizedAmount = deposits.sub(vaultMaxWithdrawAmount);
+    const vaultMinimumUncollaterizedAmount = vaultCollaterizedAmount
+      .mul(100)
+      .div(90)
+      .div(10);
```

Comment:
Isn't this supposed to be `vaultCollaterizedAmount.mul(90).div(100)` or something? To achieve 90% of vaultCollaterizedAmount

Reply:
The `vaultCollaterizedAmount` is actually the 90% locked portion, so calculate the original 10%, the formula I use is `collaterized * 100/90 * 10%`